### DISCUSSION TITLE
I'm trying to use this project in a Ai thinker audio kit 2.2 A436 (eq. 3478) with es8388 audio driver

### DISCUSSION THREAD

**Discussion by Magontek** [marcel-licence/ml_synth_sampler_example#7] [General]
I buy an ESP32 audio kit from china and found out too late the difference between the two different audio drivers in ESP32 A1S.
So now I'm trying to implement this project with the es8388 audio driver instead of the AC101.

I see you put the definition for this board in [ml_boards.h](https://github.com/marcel-licence/ML_SynthTools/blob/e7ee773d4ad68bf37251459863900ea6e43de647/src/ml_boards.h#L46) so I modify [line 65 in config.h](https://github.com/marcel-licence/ml_synth_sampler_example/blob/e39a9b03ba36200a95e3c6a09c0c0390d28849ac/config.h#L66) to `#define BOARD_ESP32_AUDIO_KIT_ES8388` but It won't compile. 

Reading the errors in compilation found out:

> Library ML SynthTools has been declared precompiled:
> Precompiled library in "C:\Users\Admin\Documents\Arduino\libraries\ML_SynthTools\src\esp32" not found

But the library ML SynthTools lib is installed so 

> Library ML SynthTools Libraries has been declared precompiled:
> Using precompiled library in C:\Users\Admin\Documents\Arduino\libraries\ML_SynthTools_Libraries\src\esp32

This didn't happened with `#define BOARD_ESP32_AUDIO_KIT_AC101` in [line 65 in config.h](https://github.com/marcel-licence/ml_synth_sampler_example/blob/e39a9b03ba36200a95e3c6a09c0c0390d28849ac/config.h#L66).

I don't know how to proceed. ¿I'm doing something wrong?

Update: Most error dissapear using the 2.0.17 version of the API for ESP32.
Buy `undefined reference to `ac101_setSourceMic()'` is stil there. I suppose it's because the AC101 it's not loading buy i'm at lost in witch is the right definition in the conditional in the macro in [line 588 ](https://github.com/marcel-licence/ml_synth_sampler_example/blob/e39a9b03ba36200a95e3c6a09c0c0390d28849ac/ml_synth_sampler_example.ino#L588).

I'm attaching the output log from the Arduino IDE of the failed compilation.
[compilation exit 1.txt](https://github.com/user-attachments/files/18129498/compilation.exit.1.txt)

**Comment by marcel-licence** [comment]
Hi @Magontek 
I've tested some audio kits with the ES8388 and using the define "#define BOARD_ESP32_AUDIO_KIT_ES8388" is the right attempt.
I've updated the code. The condition to use the ac101 functions was incorrect and should be active for the AC101 only.
Hope this brings you a step forward.
Best regards,
Marcel

**Reply by Magontek** [reply]
Thanks! It's working now.
